views/pulsar/v2/helpers/form.html.twig

```diff
@@ -2107,3 +2107,87 @@ data-*        | string | Data attributes, eg: `'data-foo': 'bar'`
+{% spaceless %}
+    <div
+        class="repeater"
+        data-repeater-add-new-group-text="{{ options['add-new-group-text']|default('Add') }}"
+        data-repeater-add-another-group-text="{{ options['add-another-group-text']|default('Add another') }}"
+        data-repeater-max-entries="{{ options['max-entries']|default(false) }}"
+    >
+        <div class="repeater__saved-data" style="display: none;"></div>
+        <div class="form__group">
+            {{
+                elem.label({
+                    'label': options.label,
+                    'required': options.required|default(false)
+                })
+            }}
+            <div class="controls">
+                <div class="table-container">
+                    <table class="table table--full repeatable__table">
+                        <thead class="repeater__preview-headings">
+                            <tr>
+                                {% for heading in options.headings %}
+                                    <th data-repeater-for-name="{{ heading.name }}">{{ heading.label }}</th>
+                                {% endfor %}
+                                <th rowspan="1" colspan="1"></th>
+                            </tr>
+                        </thead>
+                        <tbody class="repeater__preview-data">
+                            <tr class="repeater__empty-placeholder">
+                                <td colspan="{{ headingLength }}" class="muted">{{ options.empty }}</td>
+                                <td class="table__td table__td--align-right table-row__actions"></td>
+                            </tr>
+                        </tbody>
+                        <tfoot>
+                        <tr class="repeater__group" data-repeater-new-group>
+                            <td class="table__form" colspan={{ headingLength + 1 }}>
+
+{% endspaceless %}
+{% endmacro %}
+
+{% macro repeatable_end() %}
+{% import '@pulsar/pulsar/v2/helpers/html.html.twig' as html %}
+{% spaceless %}
+                                <div class="form__actions repeater__group-actions">
+                                    {{
+                                        html.button({
+                                            'data-repeater-save-group': 'true',
+                                            'label': 'Save',
+                                            'class': 'btn--primary repeater__save-group',
+                                            'type': 'button'
+                                        })
+                                    }}
+
+                                    {{
+                                        html.button({
+                                            'data-repeater-cancel-save': 'true',
+                                            'label': 'Cancel',
+                                            'class': 'btn--naked repeater__cancel-group',
+                                            'type': 'button'
```

Comment:
`type` not required

Reply:
This is required, or at least I think it is. A button without an explicit type `button` will cause a form to submit as far as I know?

Reply:
see below 👇